File: 1-Two-Sum.swift

class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var map = [Int: Int]()
        
        for i in 0..<nums.count {
            let complement = target - nums[i]
            if let complementIndex = map[complement] {
                return [complementIndex, i]
            }
            map[nums[i]] = i
        }
        
        return []
    }
}

File: 1-Two-Sum.php
class Solution {
    public function twoSum($nums, $target) {
        $map = [];
        for ($i = 0; $i < count($nums); $i++) {
            $complement = $target - $nums[$i];
            if (isset($map[$complement])) {
                return [$map[$complement], $i];
            }
            $map[$nums[$i]] = $i;
        }
        return [];
    }
}